lib/Qi/Model/Base.pm

```diff
@@ -0,0 +1,35 @@
+package Qi::Model::Base;
+use strict;
+use warnings;
+use Mojo::Base 'Mojolicious';
+use SQL::Abstract;
+
+#### self Methods ####
+has 'sql'; # $self->sql  for SQL::Abstract
+has 'pg'; # $self->pg   for Mojo::Pg
+
+sub select {
+    my ($self, $response, $table) = @_;
+    my($stmt, @bind) = $self->sql->select($table , '*');
```

Comment:
So with this as your base class, you should have a sub class Assets.pm that uses the base class (`use Mojo::Base "Qi::Model::Base";`) and then in this Assets.pm sub class, you can do `has table => 'assets';` and then you don't need to pass the table as an argument through the function.